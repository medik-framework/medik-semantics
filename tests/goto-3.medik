machine GotoMachine {
  var count;
  init state Start {
    entry {
      this.count = 0;
      goto PostStart;
    }
  }
  state PostStart {
    entry {
      if (count < 3) {
        sleep(1);
        print("Goto Machine\n");
        count = count + 1;
        goto PostStart;
      }
      exit;
    }
  }
}

init machine Driver {
  init state Start {
    entry {
      new GotoMachine();
      print("Driver Machine\n"); // Should happen before Goto's print
    }
  }
}
