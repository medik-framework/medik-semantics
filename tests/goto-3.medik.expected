Driver Machine
Goto Machine
Goto Machine
Goto Machine
